src/state.tsx

```diff
@@ -556,6 +560,15 @@ class GlobalState implements State {
     }
   }
 
+  renameLoadout(ix: number, name: string) {
+    const loadout = this.loadouts[ix];
+
+    const trimmedName = name.trim();
+    if (loadout && trimmedName) {
+      loadout.name = trimmedName;
+    }
```

Comment:
If the user clears the name by just making a blank string, should we rename it back to `Loadout {i}` instead of just ignoring the edit?

Reply:
fwiw i agree with jayden's first comment that we should do something, but i think clearing the name to a default value makes more sense

Reply:
I feel like this could lead to some unexpected behavior if the user accidentally submits the form before they have typed anything. Especially if submissions can occur whenever the input loses focus

Reply:
I think it's probably okay actually. There are other sites that implement something like this - I can't really recall from the top of my head what sites exactly - but it seems like the default behaviour being "keep the status quo" or "reset to a reasonable default" makes sense.

Even if loadout names become duplicated (someone creates "Loadout 2" and then they add another loadout), this seems more of a problem that the end user will fix if they really care enough.

Really, the goal here is to provide a very simple way of renaming loadouts without the user having to take many steps to do it. I think your UI component - with the slight adjustments it now has - achieves that quite well.

We'll probably merge this soon.